### PR TITLE
Better documented reconnecting Consumer

### PR DESCRIPTION
**Pull request by Yaytay**
A slight improvement in the resilience of Publishers and more documentation for the changes to Client.

### CODE REVIEW

**Comment by Yaytay** [comment]
Fixes #94, #91, #90

File: src/main/java/io/vertx/rabbitmq/impl/RabbitMQConsumerImpl.java [hunk 1]
```diff
@@ -92,6 +93,8 @@ public void cancel() {
   public void cancel(Handler<AsyncResult<Void>> cancelResult) {
     AsyncResult<Void> operationResult;
     try {
+      log.info("Cancelling " + consumerTag());
```
Comment (vietj) [inline src/main/java/io/vertx/rabbitmq/impl/RabbitMQConsumerImpl.java]: rather debug than info which is verbose

File: src/main/java/io/vertx/rabbitmq/RabbitMQPublisherOptions.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+   * The default connection retries = {@code Integer.MAX_VALUE}
+   */
+  public static final Integer DEFAULT_CONNECTION_RETRIES = Integer.MAX_VALUE;
+
+  /**
+   * The default internal queue size = {@code Integer.MAX_VALUE}
+   */
+  private static final int DEFAULT_QUEUE_SIZE = Integer.MAX_VALUE;
+  
+  private Integer connectionRetries = DEFAULT_CONNECTION_RETRIES;
```
Comment (vietj) [inline src/main/java/io/vertx/rabbitmq/RabbitMQPublisherOptions.java]: rather maxConnectRetries

File: src/main/java/io/vertx/rabbitmq/RabbitMQPublisherOptions.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  private long connectionRetryDelay = DEFAULT_CONNECTION_RETRY_DELAY;
```
Comment (vietj) [inline src/main/java/io/vertx/rabbitmq/RabbitMQPublisherOptions.java]: rather `reconnectInterval`

File: src/main/java/io/vertx/rabbitmq/RabbitMQPublisherOptions.java (hunk 2 above)
Comment (vietj) [reply src/main/java/io/vertx/rabbitmq/RabbitMQPublisherOptions.java]: vertx-core `NetClientOptions` use this wording

File: src/main/java/io/vertx/rabbitmq/RabbitMQPublisherOptions.java (hunk 3 above)
Comment (vietj) [reply src/main/java/io/vertx/rabbitmq/RabbitMQPublisherOptions.java]: vertx-core `NetClientOptions` use this wording

File: src/main/java/io/vertx/rabbitmq/RabbitMQConfirmListener.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.core.Handler;
+import io.vertx.core.streams.ReadStream;
+
+/**
+ *
+ * @author jtalbut
+ */
+@VertxGen
+public interface RabbitMQConfirmListener extends ReadStream<RabbitMQConfirmation> {
```
Comment (vietj) [inline src/main/java/io/vertx/rabbitmq/RabbitMQConfirmListener.java]: I don't think we need such type as it does not bring anything specific than the `ReadStream` does not provide already beside isPaused()

**Comment by vietj** [comment]
callback methods needs to have a sibling `Future<T>` method that returns the same type.

**Comment by Yaytay** [comment]
I think I've got three methods that take a Handler and don't have a corresponding sibling that returns a Future:
On RabbitMQClient:
- void addConnectionEstablishedCallback(Handler<Promise<Void>> connectionEstablishedCallback)
This can't return a Future because the callback can be called multiple times (it's not just an async operation result).

- void basicPublish(String exchange, String routingKey, BasicProperties properties, Buffer body, @Nullable Handler<Long> deliveryTagHandler, Handler<AsyncResult<Void>> resultHandler)
This should have a sibling, but doing it in the simple way causes a name clash with the basicPublish that takes a Handler<AsyncResult> but not a Handler<Long> for the delivery tag.
I could create a new name ("basicPublishWithDeliveryTag") or reorder the arguments.

On RabbitMQConsumer:
- void cancel(Handler<AsyncResult<Void>> cancelResult)
This isn't actually asynchronous any more, and is keeping the signature of the existing class for compatibility.
I can easily change the existing void cancel() to Future<Void> cancel(), or I could just admit that it's synchronous, breaking compatibility.

**Comment by Yaytay** [comment]
I've gone with renaming my basicPublish to basicPublishWithDeliveryTag, so I think I've addressed all your issues.
Thanks.

**Pull request closed without merging**